app/classifier/index.cjsx

```diff
@@ -51,8 +78,14 @@ Classifier = React.createClass
       @prepareToClassify nextProps.classification
 
     @context.geordi.remember subjectID: @props.subject?.id
+    @reCheckIfInterventionNeeded()
+
+  componentWillMount: () ->
+    @context.interventionMonitor.setProjectSlug @props.project.slug
 
   componentWillUnmount: () ->
+    @context.interventionMonitor.removeListener 'interventionRequested', => {}
```

Comment:
good spot, I will fix that. That would have been causing the listener to not deregister. 
I'm not sure it would enable me to remove the isMounted check though - as the intervention monitor can still call setState at a time when the component isn't mounted.